packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts

```diff
@@ -30,12 +30,7 @@ export const TRACE_PARENT_HEADER = 'traceparent';
 export const TRACE_STATE_HEADER = 'tracestate';
 
 const VERSION = '00';
-const VERSION_PART_COUNT = 4; // Version 00 only allows the specific 4 fields.
-
-const VERSION_REGEX = /^(?!ff)[\da-f]{2}$/;
-const TRACE_ID_REGEX = /^(?![0]{32})[\da-f]{32}$/;
-const PARENT_ID_REGEX = /^(?![0]{16})[\da-f]{16}$/;
-const FLAGS_REGEX = /^[\da-f]{2}$/;
+const TRACE_PARENT_REGEX = /^\s?((?!ff)[\da-f]{2})-((?![0]{32})[\da-f]{32})-((?![0]{16})[\da-f]{16})-([\da-f]{2})(-.*)?\s?$/;
```

Comment:
This regex is hard to understand and as a result hard to update in future.
What about building it based on on smaller pieces, e.g.
```
const VERSION_REGEX = '(?!ff)[\da-f]{2}';
const TRACE_ID_REGEX = '(?![0]{32})[\da-f]{32}';
const PARENT_ID_REGEX = '(?![0]{16})[\da-f]{16}';
const FLAGS_REGEX = '[\da-f]{2}';
const TRACE_PARENT_REGEX = new RegExp(`^\s?(${VERSION_REGEX})-(${TRACE_ID_REGEX})-(${PARENT_ID_REGEX})-(${FLAGS_REGEX})(-.*)?\s?$/;
```

Reply:
Yes, I thought so too. Splitting it up is better 👍 
I'm going for to prevent the escapes in the segments.
```js
const VERSION_REGEX = /(?!ff)[\da-f]{2}/;
const TRACE_ID_REGEX = /(?![0]{32})[\da-f]{32}/;
const PARENT_ID_REGEX = /(?![0]{16})[\da-f]{16}/;
const FLAGS_REGEX = /[\da-f]{2}/;
const TRACE_PARENT_REGEX = new RegExp(`^\\s?(${VERSION_REGEX.source})-(${TRACE_ID_REGEX.source})-(${PARENT_ID_REGEX.source})-(${FLAGS_REGEX.source})(-.*)?\\s?$/`);
```

Reply:
updated

Reply:
Thanks for the review 🤗 
The string based regex concatenation might still be cleaner despite the escapes 😁 

Should I merge everything into one commit and force push?
What do you prefer in this project?

```diff
diff --git a/packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts b/packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts
index 736981d1..8de8def9 100644
--- a/packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts
+++ b/packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts
@@ -30,12 +30,12 @@ export const TRACE_PARENT_HEADER = 'traceparent';
 export const TRACE_STATE_HEADER = 'tracestate';

 const VERSION = '00';
-const VERSION_REGEX = /(?!ff)[\da-f]{2}/;
-const TRACE_ID_REGEX = /(?![0]{32})[\da-f]{32}/;
-const PARENT_ID_REGEX = /(?![0]{16})[\da-f]{16}/;
-const FLAGS_REGEX = /[\da-f]{2}/;
+const VERSION_REGEX = '(?!ff)[\\da-f]{2}';
+const TRACE_ID_REGEX = '(?![0]{32})[\\da-f]{32}';
+const PARENT_ID_REGEX = '(?![0]{16})[\\da-f]{16}';
+const FLAGS_REGEX = '[\\da-f]{2}';
 const TRACE_PARENT_REGEX = new RegExp(
-  `^\\s?(${VERSION_REGEX.source})-(${TRACE_ID_REGEX.source})-(${PARENT_ID_REGEX.source})-(${FLAGS_REGEX.source})(-.*)?\\s?$`
+  `^\\s?(${VERSION_REGEX})-(${TRACE_ID_REGEX})-(${PARENT_ID_REGEX})-(${FLAGS_REGEX})(-.*)?\\s?$`
 );

 /**
 ```

Reply:
A few more reviews are needed before this can be merged (see https://github.com/open-telemetry/opentelemetry-js/blob/master/CONTRIBUTING.md#general-merge-requirements).

Usually it's fine to keep the branch up to date by merging master, squash happens during integration.